src/generator/synth.ts

```diff
@@ -41,13 +50,35 @@ async function main() {
   console.log(`Changes found in ${dirs.length} APIs`);
   for (const dir of dirs) {
     try {
+      const apiChangeSets = changeSets.filter(x => x.apiId.startsWith(dir));
+      const {semverity, changelog} = createChangelog(apiChangeSets);
+      let prefix: string;
+      switch (semverity) {
+        case Semverity.PATCH:
+          prefix = 'fix';
+          break;
+        case Semverity.MINOR:
+          prefix = 'feat';
+          break;
+        case Semverity.MAJOR:
+          prefix = 'feat!:';
```

Comment:
I think this won't work for `feat(templates)!: fiddled around with the templates`. Or if someone use the old approach of having `BREAKING CHANGE` in the body.

Throwing two potential ideas out there, I don't love it, but the community parser isn't [that painful to use](https://github.com/googleapis/release-please/blob/master/src/conventional-commits.ts#L175).

Alternatively, I've been threatening to write a minimal parser for about year now, because I hate how overwrought the popular one in the community is ... perhaps we should just bite the bullet and create that dependency?

Reply:
Good catch!